specs/ingestion/common/schemas/task.yml

```
@@ -266,6 +266,66 @@ SubscriptionTrigger:
 
 # Task input
 
+MappingFieldDirective:
+  type: object
+  additionalProperties: false
+  description: Describes how a field should be resolved by applying a set of directives.
+  properties:
+    fieldKey:
+      description: Destination field key.
+      type: string
+    value:
+      type: object
+      additionalProperties: true
+      description: How the destination field should be resolved from the source.
+
+MappingKitAction:
+  type: object
+  additionalProperties: false
+  description: Describes how a destination object should be resolved by means of applying a set of directives.
+  properties:
+    enabled:
+      description: Whether this action has any effect.
+      type: boolean
+    trigger:
+      description: Condition which must be satisfied to apply the field directives.
+      type: string
+    fieldDirectives:
+      type: array
+      items:
+        $ref: '#/MappingFieldDirective'
+  required:
+    - enabled
+    - trigger
+    - fieldDirectives
+
+MappingFormat:
+  type: string
+  description: Name of the mapping format schema.
+  enum: ['mappingkit/v1']
+
+MappingInput:
```

Unfortunately, I think this probably should be: 
```suggestion
InputMapping:
```

You could also just do:

```suggestion
Mapping:
```

actually. The `Input` is cumbersome, but if required, I'd probably prefer `InputMapping` to match other things in the API like `InputBigQueryOnDemand`

---

I'm just taking lead from the existing `XXXInput` types in the spec file here, rather than trying to copy the Go types from data-ingestion.

---

No problem, WFM.